Data/ByteString/Base64.hs

```diff
@@ -42,7 +42,7 @@ encode s = encodeWith Padded (mkEncodeTable alphabet) s
 -- standard that overrules RFC 4648 such as HTTP multipart mime bodies,
 -- consider using 'decodeLenient'.)
 decode :: ByteString -> Either String ByteString
-decode s = decodeWithTable Unpadded decodeFP s
+decode s = decodeWithTable Padded decodeFP s
```

Comment:
This has changed meaning. Now, I'm using the following convention: 

-  `Padded`: means that an input _must_ be padded
- `Unpadded`: means an input _must_ be unpadded
- `Don'tCare`: optionally padded or unpadded.